### DISCUSSION TITLE
socket.io-client not connecting with server after deployment

### DISCUSSION THREAD

**Discussion by subhashish12** [socketio/socket.io#4086] [Q&A]
In the local environment, it's working fine but after deployment but I'm getting error when trying to handshake with the server

`websocket.js:88 WebSocket connection to 'wss://secure-url-without-port.com/socket.io/?EIO=4&transport=websocket`

for transports i'm using 'websocket' only. socket.io-client c and socket.io on server versions are 4.2.0, on server it's running with docker and Kubernetes. Not sure what's wrong.

**Comment by darrachequesne** [comment]
Hi! Did you check https://socket.io/docs/v4/troubleshooting-connection-issues/?

**Reply by subhashish12** [reply]
i tried using curl request :
    `curl "https://server-url.com/socket.io/?EIO=4&transport=websocket" ` 
and getting respone: 
`{"code":3,"message":"Bad request"} `

I have defined transports to 'websocket' only in the server to avoid implementation of sticky sessions.

**Comment by rakNeoQOWS101** [comment]
Hi @darrachequesne,

Can you please here ? I am also facing the same issue. Also checked https://socket.io/docs/v4/troubleshooting-connection-issues/. Right now when the request is sent from the browser it fails. Following is the the client side logs. This request never reaches server. Everything works properly locally. 

`socket.io-client:url parse https://test.core.com +0ms
socket.io-client new io instance for https://test.core.com +0ms
socket.io-client:manager readyState closed +0ms
socket.io-client:manager opening https://test.core.com +0ms
engine.io-client:socket creating transport "websocket" +0ms
engine.io-client:socket options: {"path":"/websocket/","agent":false,"withCredentials":false,"upgrade":true,"jsonp":true,"timestampParam":"t","rememberUpgrade":false,"rejectUnauthorized":true,"perMessageDeflate":{"threshold":1024},"transportOptions":{},"closeOnBeforeunload":true,"autoConnect":true,"reconnectionDelay":1000,"reconnection":true,"reconnectionAttempts":null,"transports":["websocket"],"hostname":"test.core.com","secure":true,"port":"443","query":{"EIO":4,"transport":"websocket"},"socket":{"secure":true,"hostname":"test.core.com","port":"443","transports":["websocket"],"readyState":"opening","writeBuffer":[],"prevBufferLen":0,"opts":{"path":"/websocket/","agent":false,"withCredentials":false,"upgrade":true,"jsonp":true,"timestampParam":"t","rememberUpgrade":false,"rejectUnauthorized":true,"perMessageDeflate":{"threshold":1024},"transportOptions":{},"closeOnBeforeunload":true,"autoConnect":true,"reconnectionDelay":1000,"reconnection":true,"reconnectionAttempts":null,"transports":["websocket"],"hostname":"test.core.com","secure":true,"port":"443"},"id":null,"upgrades":null,"pingInterval":null,"pingTimeout":null,"pingTimeoutTimer":null}} +0ms
engine.io-client:socket setting transport websocket +1ms
socket.io-client:manager connect attempt will timeout after 20000 +2ms
socket.io-client:manager readyState opening +0ms
WebSocket connection to 'wss://test.core.com/websocket/?EIO=4&transport=websocket' failed:
engine.io-client:socket socket error {"type":"TransportError","description":{"isTrusted":true}} +2s
socket.io-client:manager error +2s
socket.io-client:manager cleanup +0ms`

We have deployed this app in k8s. Do we need to configure k8s to support it wss/ws ? Any help is appreciated.  

Thanks.,
